README.md

```diff
@@ -108,24 +108,38 @@ which is explained in detail below.
-   The workflow prints the checksum as the last step of its execution.
-6. Commit the changes to `Sources/ConcordiumWalletCrypto/crypto.swift` and `Package.swift`
-   (no other files should have changes)
-   and push an annotated tag named by the version for the new commit:
+Due to the dual nature of this repository, the release process is a little cumbersome,
+so please follow the instructions below carefully.
+
+The entire process should be done as a single PR;
+see [#11](https://github.com/Concordium/concordium-wallet-crypto-swift/pull/11) for a template.
+To preserve tags, the PR is to be merged without squashing.
+
+Steps for building and releasing a new version `<version>` of the package:
+
+1. Create and checkout a release branch named `release/<version>`.
+2. Bump the version to `<version>` in `Cargo.toml` and
+   [insert the changelog header](https://github.com/Concordium/concordium-wallet-crypto-swift/pull/11/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed).
+   Commit and push the change to the release branch.
+3. Create and push a tag named `build/<version>-<build-version>` for the new commit,
+   where `<build-version>` starts out at `0`.
+   Tagging is necessary because the workflow to be run in the next step uploads a release,
+   and GitHub requires releases to be tagged.
+4. Run the [workflow](./.github/workflows/publish-framework.yml) for publishing a new version of the binary framework.
+   Use the tag you just created as "branch" to run from and input `<version>` (i.e. without the build number).
+   If the build fails, commit a fix to the branch and return to step 3 with `<build-version>` incremented by 1.
+5. Run `make swift-bindings` locally to regenerate the Swift bridge sources.
+6. Update `Package.swift` with the updated `url` and `checksum` of the binary framework.
+   The checksum is part of the release built by the workflow above as the file `CHECKSUM`.
+7. Commit the changes to `Sources/ConcordiumWalletCrypto/generated.swift` and `Package.swift`
+   (no other files should have changes).
+   IMPORTANT: The generated Swift file is ignored by Git to prevent it from being updated untimely,
+   so it must be force added before committing:
    ```shell
-   git tag -a <version>
+   git add -f ./Sources
    ```
-   Give the tag a message describing what changed in the new version.
-
-The entire process should be done as a single PR which,
-in order to preserve the tags, gets merged without squashing.
+8. Merge PR for release branch *without squashing*.
+9. Push a tag named by the version for the merge commit on `main`.
+   The reason for tagging after merging is to avoid the release PR from being included in the autogenerated release notes
+   of the *next* (framework) release.
+   Manually create a release for this tag (without uploading any files) with a brief description of the changes from the Swift perspective.
```

Comment:
For everything to work properly we should keep pushing an annotated tag and have that trigger a workflow that does the release (using that message).